cmd/login/runner.go

```diff
@@ -181,12 +181,22 @@ func (r *runner) loginWithKubeContextName(ctx context.Context, contextName strin
 // one with the name derived from the installation code name.
 func (r *runner) loginWithCodeName(ctx context.Context, codeName string) error {
 	var contextAlreadySelected bool
+	var newLoginRequired bool
+	var selfContained bool
 
 	contextName := kubeconfig.GenerateKubeContextName(codeName)
-	err := switchContext(ctx, r.k8sConfigAccess, contextName)
+	err := switchContext(ctx, r.k8sConfigAccess, contextName, r.flag.KeepContext || (len(r.flag.SelfContained) > 0))
 	if IsContextAlreadySelected(err) {
 		contextAlreadySelected = true
 	} else if IsNewLoginRequired(err) {
+		newLoginRequired = true
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	selfContained = len(r.flag.SelfContained) > 0 && !(len(r.flag.WCName) > 0)
```

Comment:
We only create the self-contained kubeconfig here if only the MC is requested. (Otherwise it would be overwritten later when WC client cert is created)